### PR TITLE
feat(linter/unicorn): add fixer to `prefer-date-now`

### PR DESCRIPTION
**Pull request by camc314**

### CODE REVIEW

**Comment by camc314** [comment]
* **#5147** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5147?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camc314 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/unicorn/prefer_date_now.rs [hunk 1]
```diff
@@ -217,5 +223,12 @@ fn test() {
         r"function foo(){return-new Date}",
     ];
 
-    Tester::new(PreferDateNow::NAME, pass, fail).test_and_snapshot();
+    let fix = vec![
+        ("new Date().getTime()", "Date.now()"),
+        ("new Date().valueOf()", "Date.now()"),
+        ("Number(new Date())", "Date.now()"),
+        ("BigInt(new Date())", "Date.now()"),
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/unicorn/prefer_date_now.rs:230]: Can you please add tests that use semicolons, parenthesis, and type casts as well?

Comment (camc314) [reply crates/oxc_linter/src/rules/unicorn/prefer_date_now.rs:230]: done, let me know if there're any more cases you want me to add

**Comment by DonIsaac** [comment]
As a note, if you have addressed my comments, please mark the conversation as resolved and re-request my review. This way I can see it in my github notifications :)

File: crates/oxc_linter/src/rules/unicorn/throw_new_error.rs
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/unicorn/throw_new_error.rs:1]: Was this intentional, or were these changes supposed to be in another PR?

Comment (camc314) [reply crates/oxc_linter/src/rules/unicorn/throw_new_error.rs:1]: fixed

**Review by DonIsaac — APPROVED** [review]
